src/editorConfigMain.ts

```diff
@@ -198,28 +198,32 @@ function trimTrailingWhitespaceTransform(
 	textDocument: TextDocument
 ): void {
 
-	const editorAlreadyTrimsWhitespace = workspace.getConfiguration('files')['trimTrailingWhitespace'];
-	const nothingToDo = !editorconfig.trim_trailing_whitespace || editorAlreadyTrimsWhitespace;
+	const editorTrimsWhitespace = workspace.getConfiguration('files').get('trimTrailingWhitespace', false);
 
-	if (nothingToDo) {
+	if (editorTrimsWhitespace || !editorconfig.trim_trailing_whitespace) {
 		return;
 	}
 
+	const promises = [];
```

Comment:
This variable is not descriptive at all (rather generic). Please consider something like `trimmingOperations` to add clarity to the code (more human readable).